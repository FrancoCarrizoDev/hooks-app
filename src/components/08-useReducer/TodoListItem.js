import React from 'react'

export const TodoListItem = ({ todo, index, handleDeleteTask, handleToggle }) => {
    return (
        <li key={todo.id} className="list-group-item">
            <p
                className={`${todo.done ? 'complete' : ''}`}
                onClick={() => handleToggle(todo.id)}
            >
                {`${index + 1}. ${todo.desc}`}
            </p>
            <button
                className="btn btn-danger"
                onClick={() => handleDeleteTask(todo.id)}
            >
                Borrar
            </button>
        </li>
    )
}
