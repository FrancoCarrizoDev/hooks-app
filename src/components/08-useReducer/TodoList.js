import React from 'react'
import { TodoListItem } from './TodoListItem'

export const TodoList = ({todos, handleDeleteTask, handleToggle}) => {
    return (
        <div>
            <ul className="list-group list-group-flush">
                {todos.map((todo, index) => (
                    <TodoListItem todo={todo} index={index} handleDeleteTask={handleDeleteTask} handleToggle={handleToggle}/>
                ))}
            </ul>
        </div>
    )
}
