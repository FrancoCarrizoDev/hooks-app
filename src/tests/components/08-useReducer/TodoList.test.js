import { shallow } from "enzyme"
import { TodoList } from "../../../components/08-useReducer/TodoList"
import { demoTodos } from "../../fixtures/demoTodos"


describe('Test in TodoList Component', () => {
    
    const handleDeleteTask = jest.fn()
    const handleToggle = jest.fn()


    const wrapper = shallow(
        <TodoList
            todos={demoTodos}
            handleDeleteTask={handleDeleteTask}
            handleToggle={handleToggle}
        />
    )

    test('should show correctly', () => {
        
        expect(wrapper).toMatchSnapshot();
    })

    test('should be have two TodoListItem', () => {
        
        expect(wrapper.find('TodoListItem').length).toBe(demoTodos.length)

        expect(wrapper.find('TodoListItem').at(0).prop('handleDeleteTask')).toEqual(expect.any(Function))
    })
    
    
})
