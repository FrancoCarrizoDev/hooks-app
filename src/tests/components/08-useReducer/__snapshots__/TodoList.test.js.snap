// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in TodoList Component should show correctly 1`] = `
<div>
  <ul
    className="list-group list-group-flush"
  >
    <TodoListItem
      handleDeleteTask={[MockFunction]}
      handleToggle={[MockFunction]}
      index={0}
      todo={
        Object {
          "desc": "Aprender React",
          "done": false,
          "id": 1,
        }
      }
    />
    <TodoListItem
      handleDeleteTask={[MockFunction]}
      handleToggle={[MockFunction]}
      index={1}
      todo={
        Object {
          "desc": false,
          "id": 2,
        }
      }
    />
  </ul>
</div>
`;
